include/fmt/printf.h

```diff
@@ -183,6 +183,15 @@ internal::truncating_iterator<OutputIt> printf(
     basic_format_args<Context> args) {
   return Context(it, format, args).format();
 }
+
+template <typename Char, typename T>
+void format_value(buffer<Char>& buf, const T& value) {
+  formatbuf<Char> format_buf(buf);
+  std::basic_ostream<Char> output(&format_buf);
+  output.exceptions(std::ios_base::failbit | std::ios_base::badbit);
+  output << value;
+  buf.resize(buf.size());
+}
```

Comment:
Please remove this and make `loc` default to no locale in the `format_value` defined in `ostream.h`.

Reply:
I've committed the change for having only one `format_value`. In my benchmark under msvc 2019, constructing a copy of the global locale has a 9% impact on performance results. I've added an operator `bool` in `locale_ref` to avoid this.

Reply:
Good call.